src/Libraries/Microsoft.Extensions.AI.Abstractions/Utilities/AIJsonUtilities.Schema.cs

```diff
@@ -46,39 +46,47 @@ public static partial class AIJsonUtilities
     private static readonly string[] _schemaKeywordsDisallowedByAIVendors = ["minLength", "maxLength", "pattern", "format"];
 
     /// <summary>
-    /// Determines a JSON schema for the provided AI function parameter metadata.
+    /// Determines a JSON schema for the provided method signature.
     /// </summary>
+    /// <param name="methodInfo">The method from which to extract schema information.</param>
     /// <param name="title">The title keyword used by the method schema.</param>
     /// <param name="description">The description keyword used by the method schema.</param>
-    /// <param name="parameters">The AI function parameter metadata.</param>
     /// <param name="serializerOptions">The options used to extract the schema from the specified type.</param>
     /// <param name="inferenceOptions">The options controlling schema inference.</param>
     /// <returns>A JSON schema document encoded as a <see cref="JsonElement"/>.</returns>
     public static JsonElement CreateFunctionJsonSchema(
+        MethodBase methodInfo,
         string? title = null,
         string? description = null,
-        IReadOnlyList<AIFunctionParameterMetadata>? parameters = null,
         JsonSerializerOptions? serializerOptions = null,
         AIJsonSchemaCreateOptions? inferenceOptions = null)
     {
+        _ = Throw.IfNull(methodInfo);
```

Comment:
This method could previously be invoked with no arguments. Now it fails if you don't provide a methodInfo. That's by design?

Reply:
I've changed the signature of the method altogether. It didn't accept `MethodInfo` at all before this change.

Reply:
I know. And before this change, nothing was required, you could call the method with no arguments and still get back something. That's no longer the case. Was that not a scenario?

Reply:
The only scenario for this particular method was to act as an ad hoc helper for AIFunctionFactory. It used to be the case that it accepted a number of optional parameters that would populate relevant schema keywords. I experimented with keeping the same signature as before using an optional list of `ParameterInfo` but it still felt a bit ad hoc to me.

Having it take a `MethodInfo` makes it more clear that it's about generating schemas for .NET functions. It nicely complements the other public method in this class that takes a `Type` parameter.